src/include/libpmemobj/tx_base.h

```diff
@@ -69,9 +69,12 @@ enum pobj_tx_stage pmemobj_tx_stage(void);
 
 enum pobj_tx_param {
 	TX_PARAM_NONE,
-	TX_PARAM_MUTEX,	 /* PMEMmutex */
-	TX_PARAM_RWLOCK, /* PMEMrwlock */
-	TX_PARAM_CB,	 /* pmemobj_tx_callback cb, void *arg */
+	TX_PARAM_MUTEX,			/* PMEMmutex */
+	TX_PARAM_RWLOCK,		/* PMEMrwlock */
+	TX_PARAM_CB,			/* pmemobj_tx_callback cb, void *arg */
+	TX_PARAM_PREALLOC_UNDO,		/* void *addr, size_t size */
```

Comment:
Why did you call it prealloc? That might indicate that this does a preallocation. I'd just call it `TX_PARAM_SNAPSHOT_BUFFER` and `TX_PARAM_HEAP_BUFFER`